models/service_helpers.go

```diff
@@ -55,6 +55,27 @@ func checkServiceUniqueName(q *reform.Querier, name string) error {
 	}
 }
 
+func validateDBConnectionOptions(socket, host *string, port *uint16) error {
+	if host == nil && socket == nil {
+		return status.Error(codes.InvalidArgument, "Neither socket nor address passed")
+	}
+
+	if host != nil {
+		if socket != nil {
+			return status.Error(codes.InvalidArgument, "Socket and address cannot be specified together")
+		}
+
+		if port == nil {
+			return status.Errorf(codes.InvalidArgument, "Invalid field Port: value '%d' must be greater than '0'", port)
+		}
+	}
```

Comment:
Do we need to validate port range here?

Reply:
No, because uint16 range is between 0 and 65535

Reply:
In theory, we could check that port is not 0 (`pointer.GetUint16(port) == 0`), but I think it is not possible there.